### PR TITLE
[19645] ThreadSettings documentation

### PR DESCRIPTION
**Pull request by EduPonz**
This PR adds documentation for the feature provided in [Fast DDS ThreadSettings](https://github.com/eProsima/Fast-DDS/tree/feature/thread-config/main).
Depends on:

* eProsima/Fast-DDS-python#69

### CODE REVIEW

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 1]
```diff
@@ -26,66 +28,86 @@ Transport related threads (marked as UDP, TCP and SHM types) are only created wh
...
       - One per DomainParticipant
+      - ``dds.ds_ev.<participant_id>``
       - Synchronizes access to the Discovery Server |br| Database
     * - Asynchronous Writer
       - General
-      - One per enabled asynchronous |br| flow controller
+      - One per enabled asynchronous |br| flow controller.
 
         Minimum 1.
+      - ``dds.asyn.<participant_id>.<async_flow_controller_index>``
```
Comment (MiguelCompany) [inline docs/fastdds/library_overview/includes/exec_model.rst:48]: The table is cropped due to the new column. I think splitting this line (the longest in the column) might fix the problem

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 2]
```diff
@@ -102,6 +124,9 @@ but the default configuration is to always use a metatraffic port and a unicast
 This applies both to UDP and Shared Memory since TCP does not support multicast.
 More information can be found at the :ref:`listening_locators_default` page.
 
+*Fast DDS* offers the possibility of configuring certain attributes of the threads it creates by means of the
+:ref:`threadsettingsqos`.
+
```
Comment (MiguelCompany) [inline docs/fastdds/library_overview/includes/exec_model.rst:129]: Apart from this line, should we add a cross-reference to the QoS where the thread can be configured?
We could do it in the description column, with something like `Can be configured through xxxQos`

File: docs/fastdds/logging/logging_thread.rst [hunk 3]
```diff
@@ -16,6 +16,8 @@ However, sometimes applications may want to wait until the logging routine is do
 The logging module provides this capability via the member function |Log::Flush-api|.
 Furthermore, it is possible to completely eliminate the thread and its resources using member function
 |Log::KillThread-api|.
+In addition, it is possible to configured certain settings of this logging thread via the member function
```
Comment (MiguelCompany) [inline docs/fastdds/logging/logging_thread.rst:19]:
```suggestion
In addition, it is possible to configure certain settings of this logging thread via the member function
```

File: docs/fastdds/transport/shared_memory/shared_memory.rst [hunk 4]
```diff
@@ -141,36 +141,52 @@ the TransportDescriptor for Shared Memory defines the following ones:
...
+   * - ``rtps_dump_file_``
+     - ``string``
+     - ``""``
+     - |SharedMemTransportDescriptor::rtps_dump_file-api|
+     - Full path of the protocol dump file.
+   * - |PortBasedTransportDescriptor::default_reception_threads-api|
+     - |ThreadSettings|
+     -
+     - |PortBasedTransportDescriptor::default_reception_threads-api|
+     - |ThreadSettings| for the default reception threads.
```
Comment (MiguelCompany) [inline docs/fastdds/transport/shared_memory/shared_memory.rst:179]:
```suggestion
     - Default |ThreadSettings| for the reception threads.
```

Same on other transports

File: docs/fastdds/transport/shared_memory/shared_memory.rst [hunk 5]
```diff
@@ -141,36 +141,52 @@ the TransportDescriptor for Shared Memory defines the following ones:
...
+   * - |PortBasedTransportDescriptor::reception_threads-api|
+     - ``std::map<uint32_t, ThreadSettings>``
+     -
+     - |PortBasedTransportDescriptor::reception_threads-api|
+     - |ThreadSettings| for the default reception threads.
```
Comment (MiguelCompany) [inline docs/fastdds/transport/shared_memory/shared_memory.rst:184]:
```suggestion
     - |ThreadSettings| for the reception threads on specific ports.
```

Same on other transports

File: code/DDSCodeTester.cpp [hunk 6]
```diff
@@ -3945,6 +3942,9 @@ void dds_qos_examples()
         // unique for the current machine
         datasharing.automatic();
 
+        // ThreadSettings for listening thread
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:3945]:
```suggestion
        // [OPTIONAL] ThreadSettings for listening thread
```

File: code/DDSCodeTester.cpp [hunk 7]
```diff
@@ -4083,17 +4083,30 @@ void dds_qos_examples()
...
+        // Add new transport to the list of user transports
         std::shared_ptr<eprosima::fastdds::rtps::UDPv4TransportDescriptor> descriptor =
                 std::make_shared<eprosima::fastdds::rtps::UDPv4TransportDescriptor>();
         descriptor->sendBufferSize = 9126;
         descriptor->receiveBufferSize = 9126;
         transport.user_transports.push_back(descriptor);
-        //Set use_builtin_transports to false
+        // Set use_builtin_transports to false
         transport.use_builtin_transports = false;
+        // Set ThreadSettings for the builtin transports reception threads
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:4108]:
```suggestion
        // [OPTIONAL] Set ThreadSettings for the builtin transports reception threads
```

File: code/DDSCodeTester.cpp [hunk 8]
```diff
@@ -4555,6 +4571,8 @@ void dds_transport_examples ()
         udp_transport->sendBufferSize = 9216;
         udp_transport->receiveBufferSize = 9216;
         udp_transport->non_blocking_send = true;
+        udp_transport->default_reception_threads(eprosima::fastdds::rtps::ThreadSettings{2, 2, 2, 2});
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:4574]:
```suggestion

        // [OPTIONAL] ThreadSettings configuration
        udp_transport->default_reception_threads(eprosima::fastdds::rtps::ThreadSettings{2, 2, 2, 2});
```

File: code/DDSCodeTester.cpp [hunk 9]
```diff
@@ -5063,6 +5094,9 @@ void dds_usecase_examples()
         flow_control_300k_per_sec->max_bytes_per_period = 300 * 1000;
         flow_control_300k_per_sec->period_ms = 1000;
 
+        // Configure sender thread settings
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:5097]:
```suggestion
        // [OPTIONAL] Configure sender thread settings
```

File: docs/fastdds/xml_configuration/common.rst [hunk 10]
```diff
@@ -19,13 +19,16 @@ This section aims to explain these common elements.
     -   :ref:`hQos`
     -   :ref:`rLsQos`
 
+*   :ref:`ThreadSettingsType`
```
Comment (MiguelCompany) [inline docs/fastdds/xml_configuration/common.rst:22]: I think we should also add this to the index in the main XML profiles section

File: docs/fastdds/xml_configuration/domainparticipantfactory.rst [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+     - Use
+   * - ``profile_name``
+     - Sets the name under which the ``<domainparticipant_factory>`` profile is registered in the |br|
+       DDS Domain, so that it can be loaded later by the |DomainParticipantFactory-api|, as shown in |br|
+       :ref:`loadingapplyingprofiles`.
+     - Mandatory
+   * - ``is_default_profile``
+     - Sets the ``<domainparticipant_factory>`` profile as the default profile. Thus, if a default |br|
+       profile exists, it will be used when creating the DomainParticipantFactory
+     - Optional
```
Comment (MiguelCompany) [inline docs/fastdds/xml_configuration/domainparticipantfactory.rst:36]: Does it make sense for this to exist? Shouldn't it always be `true`

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: docs/fastdds/xml_configuration/domainparticipantfactory.rst (hunk 11 above)
Comment (EduPonz) [reply docs/fastdds/xml_configuration/domainparticipantfactory.rst:36]: Let's discuss this internally

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 12]
```diff
@@ -34,81 +34,95 @@ Transport related threads (marked as UDP, TCP and SHM types) are only created wh
...
-      - Synchronizes access to the Discovery Server |br| Database
+      - Synchronizes access to the Discovery Server |br| Database. |br|
+        See :ref:`dds_layer_domainParticipantQos`.
     * - Asynchronous Writer
       - General
       - One per enabled asynchronous |br| flow controller.
-
         Minimum 1.
-      - ``dds.asyn.<participant_id>.<async_flow_controller_index>``
+      - ``dds.asyn.<participant_id>.``
```
Comment (MiguelCompany) [inline docs/fastdds/library_overview/includes/exec_model.rst:49]:
```suggestion
      - ``dds.asyn.<participant_id>.`` |br|
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Review by MiguelCompany — APPROVED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Pull request merged**